src/components/SigningOfficialDaaAgreementWrapper.js

```diff
@@ -0,0 +1,101 @@
+import { h, h2, a, p, div, span } from 'react-hyperscript-helpers';
+import { Notifications } from '../libs/utils';
+import BroadLibraryCardAgreementLink from '../assets/Library_Card_Agreement_2021.pdf';
+import NIHLibraryCardAgreementLink from '../assets/NIH_Library_Card_Agreement_11_17_22_version.pdf';
+
+import { User } from '../libs/ajax';
+import Acknowledgments, {hasSOAcceptedDAAs} from '../libs/acknowledgements';
+import { useEffect, useState } from 'react';
+import { spinnerService } from '../libs/spinner-service';
+import { isNil } from 'lodash';
+
+
+
+
+export const SigningOfficialDaaAgreementWrapper = (props) => {
+  const {
+    onAccept,
+    children
+  } = props;
+
+  const [hasAccepted, setHasAccepted] = useState();
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const init = async() => {
+      try {
+        setIsLoading(true);
+        setHasAccepted(await hasSOAcceptedDAAs());
+        setIsLoading(false);
+      } catch(error) {
+        Notifications.showError({text: 'Error: Unable to retrieve current user from server'});
+        setIsLoading(false);
+      }
+    };
+    init();
+  }, []);
```

Comment:
One issue in pushing this down to the component is that other API calls are still made while this one loads which is unnecessary API traffic. Not a huge deal, but it is visible on the page. 
<img width="1431" alt="Screen Shot 2022-12-06 at 12 28 01 PM" src="https://user-images.githubusercontent.com/116679/205981057-67c79d50-8c42-4d0e-b923-7cb067e898a8.png">

Reply:
I was thinking about that too!  I'm wondering if we should flex that "Storage" muscle a bit more often.  This might be a good candidate for that.

Reply:
> I was thinking about that too! I'm wondering if we should flex that "Storage" muscle a bit more often. This might be a good candidate for that.

I agree, that's a decent approach to this problem. If the parent component made the call and set this in storage, then we could completely skip rendering unnecessary components.

Reply:
@rushtong @connorlbark - I pushed an example into the branch DUOS-2009-2 of what I was thinking around this.

Reply:
I like this - I made the logic generic, too, so it works with all future acknowledgements.